### ISSUE TITLE
Future?

### ISSUE THREAD

**Issue by fprochazka** [Kdyby/Events#97]
I have the following idea:

* move the support for lazy EventManager of Doctrine to Kdyby/Doctrine, [like symfony has it](https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php)
* this extension will only provide events mechanism for Nette (possibly rename to `kdyby/events-nette`?)
    * should we completely drop all custom code and base it of Symfony/EventDispatcher?
    * should we continue to support custom impl of EventManager?

The thing is, I don't think mixing the event mechanisms is a good idea. I think you should be able to use both systems independently, without collisions.

**Comment by enumag** [comment]
> move the support for lazy EventManager of Doctrine to Kdyby/Doctrine, like symfony has it

I disagree. This is a different case. In Symfony it's in DoctrineBridge which integrates doctrine packages into various symfony components - in this case EventManager to symfony/dependency-injection. The important note here is that DoctrineBridge only depends on doctrine/common, so unless you want to do the same with Kdyby/Doctrine it does not belong there in my opinion.

As for EventDispatcher I'm currently writing my own integration to Nette. I didn't like a few things in Symnedi/EventDispatcher (no ContainerAwareEventDispatcher, no tag for subscribers) and didn't want to BC break it so I'll deprecate it instead with a new package.

Otherwise I agree that mixing the event mechanisms is wrong. That is why I'm trying to get rid of Kdyby/Events completely in my application. What I'm aiming for is to use:

- simple ContainerAwareEventManager for Doctrine events only (no namespaces or other cool features) - implemented [here](https://github.com/Arachne/Doctrine/pull/2)
- symfony/event-dispatcher for everything else (I'll publish my package next week)
- no integration of property-based events from Nette to either of these systems
- only exception is that `Nette\Application\Application::onWhatever` will trigger appropriate events in the dispatcher (inspired by Symnedi)

**Comment by fprochazka** [comment]
> simple ContainerAwareEventManager for Doctrine events only

It could be moved to a separate package, that's a detail. My point is, that it's IMHO needed anyway and it could simplify Kdyby/Events.

**Comment by enumag** [comment]
I agree. I didn't do that because I actually used the ContainerAwareEventManager from symfony/doctrine-bridge and connected it to Nette/DI using [ContainerAdapter](https://github.com/Arachne/ContainerAdapter). If I implemented a ContainerAwareEventManager which would work with `Nette\DI\Container` directly it would be in a separate package for sure.

**Comment by enumag** [comment]
Here is my integration of EventDispatcher: https://github.com/Arachne/EventDispatcher

And I decided to put the EventManager in a separate package as well: https://github.com/Arachne/EventManager

Both packages are implemented to be as simple as possible.

**Comment by fprochazka** [comment]
So, the future:

* I'll release new Nette 2.3 and 2.4 versions
* Add lazy Doctrine EventManager to kdyby/doctrine
* deprecate this
